pkg/trace/api/normalizer_test.go

```diff
@@ -33,277 +39,315 @@ func testSpan() *pb.Span {
 	}
 }
 
+// assertNormalizationIssue asserts there is exactly one NormalizationIssue in the provided TagStats matching the reason
+func assertNormalizationIssue(t *testing.T, ts *info.TagStats, reason string) {
+	normalizationIssues := append(ts.DroppedTraceNormalizationIssues(), ts.MalformedTraceNormalizationIssues()...)
+	for _, issue := range normalizationIssues {
+		if issue.Reason == reason {
+			assert.EqualValues(t, issue.Count, 1, fmt.Sprintf("expected normalization issue %s=%d", reason, 1))
+		} else {
+			assert.EqualValues(t, issue.Count, 0, fmt.Sprintf("found unexpected normalization issue %s=%d, expecting only %s=%d", issue.Reason, issue.Count, reason, 1))
+		}
+	}
+}
+
+// assertNormalizationIssue asserts there there are no NormalizationIssues in the provided TagStats
+func assertNoNormalizationIssues(t *testing.T, ts *info.TagStats) {
+	allIssues := append(ts.DroppedTraceNormalizationIssues(), ts.MalformedTraceNormalizationIssues()...)
+	for _, issue := range allIssues {
+		assert.EqualValues(t, issue.Count, 0, fmt.Sprintf("found unexpected normalization issue %s=%d, expecting no issues", issue.Reason, issue.Count))
+	}
+}
+
 func TestNormalizeOK(t *testing.T) {
```

Comment:
If you can spare the time and don't mind too much. It would be fantastic to make one big test here called `TestNormalize` and all of the rest sub-tests of it. This is optional but would be a very much appreciated clean up.

As a side-effect, you could add the two helper functions from above inside the parent function, making it clear what their scope is, e.g.:
```go
func TestNormalize(t *testing.T) {
    assertNoIssues := func(.. ) { .... }
    assertIssue := func(...) { ... }
    
    t.Run("ok", ...
    ...
}
```

Reply:
I just tried doing it and I don't think it ends up any cleaner. 

All the tests in `normalizer_test.go` are related to normalization, and each test case is a simple independent test case. So if we grouped them all into subtests we still end up with the same amount of code, and it maybe even gets a bit more complicated due having these test cases declared dynamically vs have them be statically declared as functions. 

Use subtests might make more sense if we could statically declare these test cases in some way, like [in the go examples](https://blog.golang.org/subtests). I don't think we can do that here though.

Reply:
I understand. Ok, makes sense. Let's leave it then.